pkgs/tools/admin/colmena/default.nix

```
@@ -20,14 +20,22 @@ rustPlatform.buildRustPackage rec {
 
   NIX_EVAL_JOBS = "${nix-eval-jobs}/bin/nix-eval-jobs";
 
+  patches = [
+    # Fixes nix 2.24 compat: https://github.com/zhaofengli/colmena/pull/236
+    (fetchpatch {
+      url = "https://github.com/zhaofengli/colmena/commit/36382ee2bef95983848435065f7422500c7923a8.patch";
+      sha256 = "sha256-5cQ2u3eTzhzjPN+rc6xWIskHNtheVXXvlSeJ1G/lz+E=";
+    })
+  ];
+
   postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
     installShellCompletion --cmd colmena \
       --bash <($out/bin/colmena gen-completions bash) \
       --zsh <($out/bin/colmena gen-completions zsh) \
       --fish <($out/bin/colmena gen-completions fish)
 
     wrapProgram $out/bin/colmena \
-      --prefix PATH ":" "${lib.makeBinPath [ nixVersions.nix_2_18 ]}"
+      --prefix PATH ":" "${lib.makeBinPath [ nix ]}"
```

```suggestion
      --prefix PATH ":" "${lib.makeBinPath [ nixVersions.nix_2_24 ]}"
```

Apparently upto 2.24 is compatible, so just set it to that?

---

lets stick to the global stable version if possible

---

I would also prefer to not pin in it. The default nix version is what you get if you have nixos-rebuild and it would be weird if colmena would use something else as it's not consistent behavior.

---

Also tools should ideally not break with nix upgrade unless they link with nix as a library.